src/embedding/kd.ts

```diff
@@ -308,6 +340,110 @@ export namespace EmbeddedKpiDashboard {
         return getEventType(obj) === GdcKdCommandType.SetSize;
     }
 
+    //
+    // Add widget command
+    //
+
+    export interface IKpiWidget {
+        type: 'kpi';
+    }
+
+    export type InsightRef = { identifier: string } | { uri: string };
+
+    export interface IInsightWidget {
+        type: 'insight';
+        ref: InsightRef;
+    }
+
+    export interface IAddWidgetBody {
+        widget: IKpiWidget | IInsightWidget;
+    }
+
+    /**
+     * Adds new widget onto dashboard. New row will be created on top of the dashboard, the widget
+     * will be placed into its first column.
+     *
+     * It is currently possible to add either a KPI or an Insight. When adding either of these, KD will
+     * scroll to top so that the newly added widget is visible.
+     *
+     * For KPI, the KD will start the KPI customization flow right after the KPI is placed.
+     * Insights are placed without need for further customization
+     *
+     * Contract:
+     *
+     * -  if KD is currently editing a dashboard, then depending on widget type:
+     *    -  KPI is added to dashboard, customization flow is started, WidgetAdded will be posted
+     *    -  Insight is added to dashboard, WidgetAdded will be posted
+     *
+     * -  if insight reference included in command payload does not refer to a valid insight, CommandFailed
+     *    will be posted
+     *
+     * -  if KD is in view mode or not showing any dashboard, then CommandFailed will be posted
+     *
+     * @remarks use {@link AddWidgetCommand} to create instances of AddWidget
+     */
+    export type AddWidgetCommand = IGdcKdMessageEvent<GdcKdCommandType.AddWidget, IAddWidgetBody>;
+
+    export type AddWidgetCommandData = IGdcKdMessageEnvelope<GdcKdCommandType.AddWidget, IAddWidgetBody>;
+
+    /**
+     * Type-guard checking whether object is an instance of {@link isAddWidgetCommandData}.
+     *
+     * @param obj - object to test
+     */
+    export function isAddWidgetCommandData(obj: any): obj is AddWidgetCommandData {
+        return getEventType(obj) === GdcKdCommandType.AddWidget;
+    }
+
+    /**
+     * Adds new attribute filter to filter bar and starts the filter customization flow.
+     *
+     * Contract:
+     *
+     * -  if KD is currently editing a dashboard, adds new attribute filter, starts customization flow; FilterAdded
+     *    will be posted right after customization starts
+     *
+     * -  if KD is currently in view mode or does not show any dashboard, will post CommandFailed
+     *
+     * @remarks use {@link AddFilterCommand} factory function to instantiate
```

Comment:
`factory function` is function to create an instance of `AddFilterCommand`. But here we didn't have this kind of function. So please remove this comment.
Please check and remove all others remarks